rts/Lua/LuaUnsyncedCtrl.cpp

```diff
@@ -4864,3 +4867,57 @@ int LuaUnsyncedCtrl::Yield(lua_State* L)
 	lua_pushboolean(L, true); //hint Lua should keep calling Yield
 	return 1;
 }
+
+/* NB: strings here are never cleaned up, but the use case assumes
+ * that they live a long time and there's just a handful of them */
+std::set <std::string> tracyLuaPlots;
+
+/*** Configure custom appearence for a Tracy plot for use in debugging or profiling
+ *
+ * @function Spring.LuaTracyPlotConfig
+ * @string plotName which should be customized 
+ * @string[opt] plotFormatType "Number"|"Percentage"|"Memory", default "Number"
+ * @bool[opt] step stepwise chart, default true is stepwise
+ * @bool[opt] fill color fill, default false is no fill
+ * @number[opt] color unit32 number as BGR color, default white
+ * @treturn nil
+ */
+
+int LuaUnsyncedCtrl::LuaTracyPlotConfig(lua_State* L)
+{
+	const auto plotName             = luaL_checkstring(L, 1);
+	const auto plotFormatTypeString = luaL_optstring(L, 2, "");
+	const auto step                 = luaL_optboolean(L, 3, true); // stepwise default
+	const auto fill                 = luaL_optboolean(L, 4, false); // no fill default
+	const uint32_t color            = luaL_optint(L, 5, 0xFFFFFF); // white default
+
+	tracy::PlotFormatType plotFormatType;
+	switch (plotFormatTypeString[0]) {
+		case 'p': case 'P': plotFormatType = tracy::PlotFormatType::Percentage; break;
+		case 'm': case 'M': plotFormatType = tracy::PlotFormatType::Memory;     break;
+		default:            plotFormatType = tracy::PlotFormatType::Number;     break;
+	}
+
+	const auto [iterator, inserted] = tracyLuaPlots.emplace(plotName);
```

Comment:
If we can't get stable string pointer from lua, then:

Declaring the `tracyLuaPlots` as `std::set<std::string, std::less<>>` so that lookup doesn't need to construct temporary string and then doing:

```
auto it = tracyLuaPlots.find(plotName);
if (it == tracyLuaPlots.end()) {
  it = tracyLuaPlots.emplace(plotName).first;
}
```

will be a bunch of times quicker, I did some microbenchmarking for fun: https://gist.github.com/p2004a/c045a896a567074c40120eb902d86a16

Reply:
Nice. Seems to be missing a benchmark with `std::less<>` and `emplace` though.

Reply:
I've just run it just to be sure and it's 100% equivalent to the current code. That's because when using emplace with `std::less<>`, it still needs to construct something to use it as a key lookup, it only takes the constructor arguments not key itself, looking at the code, it basically constructs temporary RB tree node https://github.com/gcc-mirror/gcc/blob/master/libstdc%2B%2B-v3/include/bits/stl_tree.h#L2422-L2435

Reply:
@p2004a based on your results, do you have a suggested change? I dont know enough about the std lib to make a sane change :(

Reply:
It boils down to the exact code snippet @p2004a posted in his comment. I pushed a commit onto the branch